gufe/protocols/protocolunit.py

```diff
@@ -30,26 +31,24 @@ class Context:
     shared: PathLike
 
 
-class ProtocolUnitMixin:
-
-    def _list_dependencies(self, cls):
-        deps = [] 
-        for key, value in self.inputs.items():
-            if isinstance(value, dict):
-                for k, v in value.items():
-                    if isinstance(v, cls):
-                        deps.append(v)
-            elif isinstance(value, list):
-                for i in value:
-                    if isinstance(i, cls):
-                        deps.append(i)
-            elif isinstance(value, cls):
-                deps.append(value)
-        return deps
-
-
-class ProtocolUnitResultBase(GufeTokenizable, ProtocolUnitMixin):
-    def __init__(self, *, 
+def _list_dependencies(inputs, cls):
```

Comment:
@dwhswenson it might cause you grief in a merge conflict, but I got rid of this mixin since it didn't make sense to unite ProtocolUnit and Result with a common ancestor, so we can just have a function instead

Reply:
Talk to @dotsl on that -- he had a reason to do this, but I don't remember the argument.

Reply:
(And I accidentally notified the wrong person ... that was supposed to be @dotsdl (fixed now); sorry dotSL)